File: onlinebanking-facade/src/main/java/domain/Contract.java
package domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Created by alexg on 18.05.17.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Contract {

    private String logo;
    private String homepage;
    private String hotline;
    private String email;
    private String mandateReference;
    private Cycle interval;
    private boolean cancelled;
}

File: multibanking-app/src/api/Contract.ts
export interface Contract {
    "email"?: string;
    "homepage"?: string;
    "hotline"?: string;
    "interval"?: Contract.IntervalEnum;
    "logo"?: string;
    "amount"?: number;
    "mandateReference"?: string;
}

export namespace Contract {
    export enum IntervalEnum {
        WEEKLY = <any> 'WEEKLY',
        MONTHLY = <any> 'MONTHLY',
        QUARTERLY = <any> 'QUARTERLY',
        HALFYEARLY = <any> 'HALFYEARLY',
        YEARLY = <any> 'YEARLY'
    }
}
